src/connection_status.rs

```diff
@@ -112,7 +112,10 @@ impl ConnectionStatus {
     }
 
     pub(crate) fn poison(&self, err: Error) {
-        self.lock_inner().poison(err);
+        if let Some((resolver, _connection)) = self.lock_inner().poison(err.clone()) {
```

Comment:
This pins the lock lifetime to the if, so it won't work

```
        let resolver = self.lock_inner().poison(err.clone());
        if let Some((resolver, _connection)) = resolver {
```
This will work, as it pins the lifetime of the lock on the let assignment, then the connection to the if

Reply:
hrm, had thought since the `lock_inner()` call was a tmp and only the result of `.poison()` was bound to a variable, it'd be dropped

currently waiting on test build to try

Reply:
ok, minimal repro is holding steady going back to my application now